R/aggregate_results.R

```diff
@@ -16,23 +16,31 @@
 #'   arguments.
 #' @param iter_var String vector holding the variable over which to iterate in
 #'   sensitivity analysis
+#' @param risk_type String that is either lrisk or trisk
 #'
 #' @return A list including basic and aggregated results.
-aggregate_results <- function(results_list, sensitivity_analysis_vars, iter_var) {
+aggregate_results <- function(results_list, sensitivity_analysis_vars, iter_var, risk_type) {
   sensitivity_analysis_vars <- paste0(sensitivity_analysis_vars, "_arg")
 
   # Aggregate Crispy Results -----
 
+  merge_by_cols <- c(
+    "investor_name", "portfolio_name", "scenario_name", "scenario_geography",
+    "company_name", "ald_sector", "asset_type_arg", "baseline_scenario_arg",
+    "shock_scenario_arg", "lgd_arg", "risk_free_rate_arg", "discount_rate_arg",
+    "growth_rate_arg", "div_netprofit_prop_coef_arg", "shock_year_arg",
+    "fallback_term_arg", "use_company_terms_arg"
+  )
+
+  if (risk_type == "lrisk") {
+    lrisk_additional <- c("scc_arg", "settlement_factor_arg", "exp_share_damages_paid_arg")
+    merge_by_cols <- append(merge_by_cols, lrisk_additional)
```

Comment:
one more tiny nit: instead of append you can simply use c to combine to vectors, append is more typically used for lists.